com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/ProcessMtaArchiveStep.java

```diff
@@ -93,6 +107,64 @@ private void processApplicationArchive(final DelegateExecution context, String a
                 getStepLogger().debug("MTA Archive Resources: {0}", mtaArchiveResources.keySet());
             });
         fileService.processFileContent(manifestProcessor);
+
+        setResourcesParameters(context, appArchiveId);
+    }
+
+    private void setResourcesParameters(DelegateExecution context, String appArchiveId) throws FileStorageException {
+        DeploymentDescriptor deploymentDescriptor = StepsUtil.getUnresolvedDeploymentDescriptor(context);
+
+        int majorVersion = getMajorVersion(deploymentDescriptor);
+        PropertiesAccessor propertiesAccessor = new HandlerFactory(majorVersion).getPropertiesAccessor();
+
+        for (Resource resource : deploymentDescriptor.getResources1()) {
+            String resourceFileName = StepsUtil.getResourceFileName(context, resource.getName());
+            if (resourceFileName != null) {
+                String reletedToResourceFileContent = getReletedToResourceFileContent(context, appArchiveId, resourceFileName);
+                Map<String, Object> relatedResourceFileContentMap = parseRelatedResourceFileContent(reletedToResourceFileContent);
+                Map<String, Object> currentResourceParameters = propertiesAccessor.getParameters(resource);
+                Map<String, Object> resourceConfigParameters = createConfigParameters(relatedResourceFileContentMap);
+                Map<String, Object> mergedResourceParameters = mergeResourceParameters(currentResourceParameters,
+                    resourceConfigParameters);
+                propertiesAccessor.setParameters(resource, mergedResourceParameters);
+            }
+        }
+        StepsUtil.setUnresolvedDeploymentDescriptor(context, deploymentDescriptor);
+    }
+    
+    private Map<String, Object> createConfigParameters(Map<String, Object> resourceParameters) {
+       Map<String, Object> configParameters = new HashMap<>();
+       configParameters.put(SupportedParameters.SERVICE_CONFIG, resourceParameters);
+       
+       return configParameters;
+    }
+
+    private int getMajorVersion(DeploymentDescriptor deploymentDescriptor) {
+        String schemaVersion = deploymentDescriptor.getSchemaVersion();
+
+        return Integer.valueOf(schemaVersion.split("\\.")[0]);
+    }
+
+    private Map<String, Object> parseRelatedResourceFileContent(String reletedToResourceFileContent) {
+        return JsonUtil.convertJsonToMap(reletedToResourceFileContent);
+    }
+
+    private Map<String, Object> mergeResourceParameters(Map<String, Object> currentParameters, Map<String, Object> newParameters) {

```

Comment:
Handle situation when currentParameters are NULL.
This was done in CreateOrUpdateServicesStep: mergeCredentials